omnibus/files/private-chef-ctl-commands/rotate_credentials.rb

```diff
@@ -117,11 +145,94 @@
 add_command_under_category "show-service-credentials", "credential-rotation", "Show the service credentials", 2 do
   ensure_configured!
 
-  credentials = Veil::CredentialCollection::ChefSecretsFile.from_file("/etc/opscode/private-chef-secrets.json")
+  credentials = Veil::CredentialCollection::ChefSecretsFile.from_file(secrets_file_path)
   pp(credentials.legacy_credentials_hash)
   exit(0)
 end
 
+# require-credential-rotation is designed to put the Chef Server in a state
+# where it's offline and requires complete credential rotation to restart.
+# This sort of thing is useful if you're creating public Chef Server images
+# and you want to make sure that the Chef Server doesn't start up until all
+# secrets have been rotated. It's important to note that credential rotation
+# does not rotate the pivotal, user or client keys, or remove any Chef Server
+# policy or cookbooks that have been uploaded.
+add_command_under_category "require-credential-rotation", "credential-rotation", "Disable the Chef Server and require credential rotation", 2 do
+  @agree_to_disable = false
+  @ui = HighLine.new
+
+  OptionParser.new do |opts|
+    opts.banner = "Usage: chef-server-ctl require-credential-rotation [--yes]"
+
+    opts.on("-h", "--help", "Show this message") do
+      puts opts
+      exit
+    end
+
+    opts.on("-y", "--yes", "Agree to disable the Chef Server and require credential rotation") do
+      @agree_to_disable = true
+    end
+  end.parse!(ARGV)
+
+  # Agree to disable and require rotation
+  msg = "Are you sure you want to disable the Chef Server and require credential rotation?" \
+          "\n Type 'yes' to confirm: "
+  exit(1) unless @agree_to_disable || @ui.ask("<%= color(%Q(#{msg}), :yellow) %>") =~ /yes/i
+
+  # Shut down Chef Server
+  run_sv_command("stop")
+
+  # Disable services
+  get_all_services.each do |service|
+    File.unlink("#{service_path}/#{service}") if service_enabled?(service)
+  end
+
+  # Create global pre hook that'll prevent ctl commands from running until the
+  # credentials have been rotated
+  File.open(credential_prehook_file_path, "a+") do |file|
+    file.write <<-EOF.gsub(/^\s{6}/, "")
+      add_global_pre_hook "require_credential_rotation" do
+        # Allow running "chef-server-ctl rotate-shared-secrets"
+        # "chef-server-ctl" is a wapper that runs "omnibus-ctl opscode $command"
+        # so we'll look for our command and opscode in the args
+
+        return true if ARGV.include?("opscode") && ARGV.include?("rotate-shared-secrets")
```

Comment:
This check would make it possible to do something like `chef-server-ctl start oc_erchef rotate-shared-secrets`.  I don't know how much a concern that is.